modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/solace/SolacePublisherPortalTestCase.java

```diff
@@ -0,0 +1,943 @@
+        additionalPropertiesObj.put("provider", user.getUserName());
+        additionalPropertiesObj.put("name", solaceApi1Name);
+        additionalPropertiesObj.put("context", solaceApi1Context);
+        additionalPropertiesObj.put("version", solaceApiVersion);
+        additionalPropertiesObj.put("gatewayVendor", solaceGatewayVendor);
+        additionalPropertiesObj.put("type", "WEBSUB");
+
+        String solaceDefinitionPath = FrameworkPathUtil.getSystemResourceLocation() + "solace"
+                + File.separator + "APIMaintenance.yml";
+        File file = new File(solaceDefinitionPath);
+
+        // Validate Async Definition of the Solace Specification
+        AsyncAPISpecificationValidationResponseDTO asyncAPISpecificationValidationResponseDTO = restAPIPublisher.
+                validateAsyncAPISchemaDefinition(null, file);
+        assertTrue(asyncAPISpecificationValidationResponseDTO.isIsValid());
+
+        // Create API by importing the Solace  definition
+        APIDTO solaceApiDto = restAPIPublisher
+                .importAsyncAPISchemaDefinition(file, null, additionalPropertiesObj.toString());
+
+        // Make sure API is created properly
+        solaceApi1Id = solaceApiDto.getId();
+        assertEquals(solaceApiDto.getName(), solaceApi1Name);
+        assertEquals(solaceApiDto.getContext(), "/" + solaceApi1Context);
+
+        // Assert Solace specific API properties
+        assertEquals(solaceApiDto.getGatewayVendor(), solaceGatewayVendor);
+        List<String> asyncProtocolsForAssertion = new ArrayList<String>();
+        asyncProtocolsForAssertion.add("mqtt");
+        asyncProtocolsForAssertion.add("http");
+        assertEquals(solaceApiDto.getAsyncTransportProtocols(), asyncProtocolsForAssertion);
+    }
+
+    @Test(groups = {"wso2.am"}, description = "Deploy created Solace API into Solace broker",
+            dependsOnMethods = "testSolaceDefinitionImport")
+    public void testSolaceAPIDeployToSolaceBroker() throws Exception {
+
+        log.info("testSolaceAPIDeployToSolaceBroker initiated");
+
+        // Create revision and deploy to Solace broker
+        createSolaceAPIRevisionAndDeployToSolaceBroker(solaceApi1Id, restAPIPublisher);
+        restAPIPublisher.changeAPILifeCycleStatusToPublish(solaceApi1Id, false);
+        waitForAPIDeploymentSync(user.getUserName(), solaceApi1Name, solaceApiVersion,
+                APIMIntegrationConstants.IS_API_EXISTS);
+    }
+
+    @Test(groups = {"wso2.am"}, description = "Undeploy deployed Solace API from Solace broker",
+            dependsOnMethods = "testSolaceAPIDeployToSolaceBroker")
+    public void testSolaceAPIUndeployFromSolaceBroker() throws Exception {
+
+        log.info("testSolaceAPIUndeployFromSolaceBroker initiated");
+
+        //Undeploy from Solace broker
+        undeployAndDeleteSolaceAPIRevisionsFromSolaceBroker(solaceApi1Id, restAPIPublisher);
+        waitForAPIUnDeploymentSync(user.getUserName(), solaceApi1Name, solaceApiVersion,
```

Comment:
arnt we going to assert something here to check whether its undeployed ?

Reply:
Fixed with https://github.com/wso2/product-apim/pull/11970/commits/576d827cdbee9c8ff0fb7764f47259ea436deac8